mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,143 @@
+    assert np.allclose(orig_data[n_channels // 2, 150:] - 100, np.arange(150))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+
+
+@requires_testing_data
+def test_nsx_ver_22():
+    """Primary tests for NSx reader"""
+    raw = read_raw_nsx(nsx_22_fname,)
+    assert getattr(raw, '_data', False) is False
+    assert raw.info['sfreq'] == 2000
+
+    # Check info object
+    assert raw.info['meas_date'].day == 31
+    assert raw.info['meas_date'].year == 2023
+    assert raw.info['meas_date'].month == 1
+    assert raw.info['chs'][0]['cal'] == 0.6103515625
+    assert raw.info['chs'][0]['range'] == 0.001
+
+    # check raw_extras
+    for r in raw._raw_extras:
+        assert r['orig_format'] == raw.orig_format
+        assert r['orig_nchan'] == 128
+        assert len(r['timestamp']) == len(r['nb_data_points'])
+        assert len(r['timestamp']) == len(r['offset_to_data_block'])
+
+    # Check annotations
+    assert len(raw.annotations) == 0
+
+    raw.load_data()
+    raw_data, times = raw[:]
+    n_channels, n_times = raw_data.shape
+    assert times.shape[0] == n_times
+    orig_data = raw_data / (raw.info['chs'][0]['cal']
+                            * raw.info['chs'][0]['range'])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+
+
+@requires_testing_data
+def test_stim_eog_misc_chs_in_nsx():
+    raw = read_raw_nsx(nsx_22_fname, stim_channel='elec127', eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=['elec127'], eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=127, eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=[127], eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    stims = [ch_info['kind'] == FIFF.FIFFV_STIM_CH for ch_info
+             in raw.info['chs']]
+    assert np.any(stims)
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    try:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=['elec128', 129],
+                           eog=['elec126'])
+    except ValueError as err:
+        err_msg = err.args[0]
+    assert err_msg == 'Invalid stim_channel'
```

Comment:
you want to use `pytest.raises` ... in general avoid `try`/`except` pattern when you can. It will silence errors

Reply:
Thanks for pointing that out.